### PR TITLE
Feat/persona score

### PR DESCRIPTION
**Pull request by VirAlex**
## Objective
What: Port the feature on Lululemon to display Personalisation Strat into the demo guide
Why: Because, we thought it could be really helpful for SEs to display live on front end the impact of each Persona Strat.
How: I fetch the API to get the answer and I rendered it dynamic, with the size of Bars, colors, title,...

## Type

- [x] New Feature

## Tested

✅

## Documented

- [x] Have you documented in changed file using comments

Screenshot 2022-08-03 at 16 43 44
Screenshot 2022-08-03 at 16 43 53

### CODE REVIEW

File: src/components/appliedRules/PersonaScore.jsx [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+    hierarchicalCategoriesLvl0,
+    hierarchicalCategoriesLvl1,
+    hierarchicalCategoriesLvl2,
+    hierarchicalCategoriesLvl3,
+    genderFilter,
+    category,
+    brand,
+    categories,
+  } = hitsConfig;
+  const [colourArray, setColourArray] = useState([]);
```
Comment (HugoWith) [inline src/components/appliedRules/PersonaScore.jsx:18]: We might have the possibility to create one state that contain an object : 

`const [setObject, object] = useState({
color: [],
brand: [],
... etc
})` 

Maybe I'm wrong 😊

Comment (VirAlex) [reply src/components/appliedRules/PersonaScore.jsx:18]: Yes I think you're right it's just with the step to get data from the array if it's not empty push the new value to existant values.
I don't know if we can do it:
```suggestion
          if (colourArray.length > 0) {
          setColourArray([...colourArray, resultsScore.colour]);
        }
```

File: src/components/appliedRules/PersonaScore.jsx [hunk 2]
```diff
@@ -15,14 +15,7 @@ const PersonaScore = (props) => {
...
-  // const [genderArray, setGenderArray] = useState([]);
-  // const [hierarchical0Array, sethierarchical0Array] = useState([]);
-  // const [hierarchical1Array, sethierarchical1Array] = useState([]);
-  // const [hierarchical2Array, sethierarchical2Array] = useState([]);
-  // const [categoryArray, setCategoryArray] = useState([]);
-  // const [brandArray, setBrandArray] = useState([]);
-  // const [categoriesArray, setCategoriesArray] = useState([]);
+
   const [scorePersona, setScorePersona] = useState({
     color: [],
```
Comment (VirAlex) [inline src/components/appliedRules/PersonaScore.jsx:20]: Done :✅ @HugoWith

File: src/components/appliedRules/PersonaScore.jsx [hunk 3]
```diff
@@ -52,10 +56,101 @@ const PersonaScore = (props) => {
...
+      color: [],
+      brand: [],
+      gender: [],
+      hierarchical0: [],
+      hierarchical1: [],
+      hierarchical2: [],
+      category: [],
+      categories: [],
+    };
+    if (!isSwitchToggle) {
```
Comment (VirAlex) [inline src/components/appliedRules/PersonaScore.jsx:72]: Add the fact than if we used the personaFilters at the place of the event sending we can switch the button, by default it's true
@matthew1809

**Comment by HugoWith** [comment]
Thanks @VirAlex . Could I ask you to add more space between each rules that we're showing. It's a bit too crowded I think. What's your opinion on that ? 


Screenshot 2022-08-08 at 14 34 20
n

**Comment by VirAlex** [comment]
> Thanks @VirAlex . Could I ask you to add more space between each rules that we're showing. It's a bit too crowded I think. What's your opinion on that ?
> 
> Screenshot 2022-08-08 at 14 34 20
> 
> n

Yes you're right I'll do it 🙏

File: src/components/appliedRules/SCSS/appliedRules.scss [hunk 4]
```diff
@@ -68,11 +67,13 @@
     flex-direction: column;
     gap: 0.5rem;
     margin-top: 0.5rem;
+    margin-bottom: 0.5rem;
```
Comment (VirAlex) [inline src/components/appliedRules/SCSS/appliedRules.scss:70]: Better ? @HugoWith 
Screenshot 2022-08-08 at 14 47 19

Comment (HugoWith) [reply src/components/appliedRules/SCSS/appliedRules.scss:70]: Wonderful ✅

**Review by HugoWith — APPROVED** [review]
APPROVED

**Pull request merged**